flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerBase.java

```diff
@@ -164,6 +168,8 @@
     private final Map<OperatorID, OperatorCoordinatorHolder> coordinatorMap;
 
     private final ComponentMainThreadExecutor mainThreadExecutor;
+    private final Map<ExecutionAttemptID, ErrorInfo> localTaskFailures = new HashMap<>();
+    private final Map<Set<ExecutionAttemptID>, ErrorInfo> globalTaskFailures = new HashMap<>();
```

Comment:
For what do we need the maps here? Wouldn't a single `List<ErrorInfo>` be good enough here?

Reply:
You're right. That was still some leftover of my initial work considering multiple exceptions per failure cause. I'm gonna refactor it to use a List here in order to be more consistent.